### PR TITLE
feat: throw ExitException instead of hard exit

### PR DESCRIPTION
**Pull request by pascalgn**
I replaced `System.exit` calls I found with `throw new ExitException()` so that it's easier to use this library in our code base :)

I also made some minor cleanups, if you want I can add them to another PR of course.

### CODE REVIEW

**Comment by havardh** [comment]
Great! Thank you :)

Could you re-word the commit message with the prefix `feat`? This project uses the angular ([commit convention](https://gist.github.com/stephenparish/9941e89d80e2bc58a153)), and a `refactor` should not contain any functional changes. :+1:

I'd be happy to hear about how you plan to use it in your code base, and feel free to let me know if you find any other rough edges :)

File: src/main/java/com/github/havardh/javaflow/exceptions/ExitException.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.github.havardh.javaflow.exceptions;
+
+public class ExitException extends RuntimeException {
+  private final int code;
+
+  public ExitException(int code, Throwable cause) {
+    super(cause);
+    this.code = code;
```
Comment (havardh) [inline src/main/java/com/github/havardh/javaflow/exceptions/ExitException.java]: I might suggest that we make the code an `enum` which tells us what the actual cause was with individual exit codes. 

```
enum ErrorCode {
  TYPE_MAP_NOT_FOUND(1),
  COULD_NOT_PARSE_SOURCE_CODE(2),
  // ...

  private int code;
  public ErrorCode(int code) { this.code = code; }
  
}

```

Comment (pascalgn) [reply src/main/java/com/github/havardh/javaflow/exceptions/ExitException.java]: Good idea, fixed! 👍

**Comment by pascalgn** [comment]
I changed the commit message to `feat:` prefix! 👍

I was trying to use the library with our project, but `mvn generate-sources` kept exiting with error code 1, so I was a bit confused, that's where this change is coming from :)

**Comment by havardh** [comment]
Great I'll see if I get time to test it and merge it this weekend :)

Are you using https://github.com/havardh/javaflow-maven-plugin to integrate it into your maven build?

**Comment by pascalgn** [comment]
Yes exactly, I'm using https://github.com/havardh/javaflow-maven-plugin!

File: src/main/java/com/github/havardh/javaflow/phases/reader/FileReader.java [hunk 2]
```diff
@@ -23,10 +25,8 @@
     try {
       return of(new String(Files.readAllBytes(Paths.get(filename))));
     } catch (IOException e) {
-      e.printStackTrace();
-      System.exit(1);
+      throw new ExitException(ErrorCode.COULD_NOT_PARSE_SOURCE_CODE, e);
```
Comment (havardh) [inline src/main/java/com/github/havardh/javaflow/phases/reader/FileReader.java]: The `FileReader` does not parse, so I propose to make this `ErrorCode.COULD_NOT_READ_FILE`.

Comment (pascalgn) [reply src/main/java/com/github/havardh/javaflow/phases/reader/FileReader.java]: Good point, fixed!

**Pull request merged**